lib/Twig/Parser.php

```diff
@@ -328,17 +328,30 @@ private function filterBodyNodes(Twig_Node $node)
             throw new Twig_Error_Syntax('A template that extends another one cannot include content outside Twig blocks. Did you forget to put the content inside a {% block %} tag?', $node->getTemplateLine(), $this->stream->getSourceContext());
         }
 
-        // bypass nodes that will "capture" the output
+        // bypass nodes that "capture" the output
         if ($node instanceof Twig_NodeCaptureInterface) {
+            // a "block" tag in such a node will serve as a block definition AND be displayed in place as well
             return $node;
         }
 
+        // "block" tags that are not captured (see above) are only used for defining
+        // the content of the block. In such a case, nesting it does not work as
+        // expected as the definition is not part of the default template code flow.
+        if ($nested && $node instanceof Twig_Node_BlockReference) {
+            //throw new Twig_Error_Syntax('A block definition cannot be nested under non-capturing nodes.', $node->getTemplateLine(), $this->stream->getSourceContext());
+            @trigger_error('Nesting a block definition under a non-capturing node is deprecated since version 2.5.0 and will become a syntax error in 3.0.', E_USER_DEPRECATED);
```

Comment:
should it mention the template name, to know where this can be fixed ? the 3.x exception will have the location, but this message does not

Reply:
done